bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerHandle.java

```diff
@@ -238,15 +238,27 @@ public Integer getSample() {
                     return pendingAddOps.size();
                 }
             });
-        initializeExplicitLacFlushPolicy();
+
+        initializeWriteHandleState();
+    }
+
+    protected void initializeWriteHandleState() {
+        if (!getLedgerMetadata().isClosed()
+            && !(this instanceof ReadOnlyLedgerHandle)
```

Comment:
This is weird, can't we avoid?

Reply:
Sure, might as well clean it up now that it doesn't get called in the other path.

Reply:
Seems to me that you didn't 'push'